test/core/end2end/cq_verifier.cc

```diff
@@ -269,21 +269,36 @@ std::string CqVerifier::ToShortString() const {
 }
 
 void CqVerifier::FailNoEventReceived(const SourceLocation& location) const {
-  fail_(Failure{location, "No event received", ToStrings()});
+  fail_(Failure{location, "No event received", ToStrings(), {}});
 }
 
 void CqVerifier::FailUnexpectedEvent(grpc_event* ev,
                                      const SourceLocation& location) const {
+  std::vector<std::string> message_details;
+  if (ev->type == GRPC_OP_COMPLETE && ev->success) {
+    auto successful_state_strings = successful_state_strings_.find(ev->tag);
+    if (successful_state_strings != successful_state_strings_.end()) {
+      for (SuccessfulStateString* sss : successful_state_strings->second) {
+        message_details.emplace_back(sss->GetSuccessfulStateString());
```

Comment:
Style nit: https://abseil.io/tips/112. Prefer `push_back` when `emplace_back` would do the same thing.

Reply:
ok for here, however I'd push back fairly strongly on that for our mainline code.